libtrap/src/ifc_file.c

```diff
@@ -109,6 +110,28 @@ void file_terminate(void *priv)
 }
 
 /**
+ * \brief Create path, recursive.
+ * \param[in] path where file will be created
+*/
+static void _mkdir(const char *dir) {
```

Comment:
This function can be improved a lot.
1) If you change the function declaration to  _mkdir(char *dir), you don't need additional buffer and just iterate over
    the received pointer. It is much safer than the segfault you created on line 123. 
    BTW: snprintf() already returns the number of printed characters
2) Why 256 bytes? Just check whether the path is longer than PATH_MAX - 1 from limits.h.
    Now you have two options... 
         a) either raise a warning and truncate anything past the PATH_MAX - 1 characters (set last char to '\0')
         b) raise an error
   In both cases your function should return int instead of void to indicate, whether the function ended up creating the path or not.
3) You MUST check the return value of mkdir() on line 129 and create according behavior.

PS: Next time you copy someone else's code, make sure the code makes sense, it has no flaws and it cannot be done better.

Reply:
Just a comment on using 3rd party code - The original source MUST be cited! NEMEA is an open source project and we don't want to violate copyrights!!!